drivers/wifi/infineon/airoc_wifi.h

```diff
@@ -6,16 +6,48 @@
  */
 
 #include <whd_buffer_api.h>
-#include <zephyr/sd/sd.h>
-#include <zephyr/sd/sdio.h>
 #include <zephyr/drivers/gpio.h>
 #include <zephyr/net/wifi_mgmt.h>
+
+#ifdef CONFIG_AIROC_WIFI_BUS_SDIO
+#include <zephyr/sd/sd.h>
+#include <zephyr/sd/sdio.h>
+#endif
+#ifdef CONFIG_AIROC_WIFI_BUS_SPI
+#include <hardware/gpio.h>
+#include <zephyr/drivers/spi.h>
+#endif
+
 #include <cy_utils.h>
 
+#if defined(CONFIG_AIROC_WIFI_BUS_SPI)
+#define AIROC_WIFI_SPI_OPERATION (SPI_WORD_SET(DT_PROP_OR(DT_DRV_INST(0), spi_word_size, 8)) \
+			| (DT_PROP(DT_DRV_INST(0), spi_half_duplex) ? SPI_HALF_DUPLEX : SPI_FULL_DUPLEX) \
+			| SPI_TRANSFER_MSB)
+#endif
+
+#if defined(CONFIG_AIROC_WIFI_BUS_SPI_DATA_IRQ_MUX) && defined(CONFIG_BOARD_RPI_PICO)
+#define PICO_BUS_DEVICE DT_PARENT(DT_PARENT(DT_DRV_INST(0)))
+
+#if PICO_BUS_DEVICE == DT_NODELABEL(pio0)
```

Comment:
This is very strange and violates the independence of the modules.
Can't you put it on the pio-spi side?